client/consensus/beefy/src/tests.rs

```diff
@@ -1263,18 +1263,35 @@ async fn beefy_reports_equivocations() {
 
 	let peers = peers.into_iter().enumerate();
 	// finalize block #1 -> BEEFY should not finalize anything (each node votes on different MMR).
-	finalize_block_and_wait_for_beefy(&net, peers, &hashes[1], &[]).await;
+	let (best_blocks, versioned_finality_proof) = get_beefy_streams(&mut net.lock(), peers.clone());
+	peers.clone().for_each(|(index, _)| {
+		let client = net.lock().peer(index).client().as_client();
+		client.finalize_block(hashes[1], None).unwrap();
+	});
+
+	// run for up to 5 seconds waiting for Alice's report of Bob/Bob_Prime equivocation.
+	for wait_ms in [250, 500, 1250, 3000] {
+		run_for(Duration::from_millis(wait_ms), &net).await;
+		let alice_reported_equivocations =
+			api_alice.reported_equivocations.as_ref().unwrap().lock();
+		if alice_reported_equivocations.is_empty() {
+			continue
```

Comment:
If I understand correctly, If after the last retry `alice_reported_equivocations` is still empty, we break out of the loop and we don't get to verify the expected equivocation. Is this desired ?

Reply:
right, good catch! I fixed it, ptal